datafusion/core/src/physical_optimizer/optimizer.rs

```diff
@@ -79,6 +80,8 @@ impl PhysicalOptimizer {
             // repartitioning and local sorting steps to meet distribution and ordering requirements.
             // Therefore, it should run before EnforceDistribution and EnforceSorting.
             Arc::new(JoinSelection::new()),
+            // The LimitedDistinctAggregation rule should be applied before the EnforceDistribution rule
```

Comment:
I think adding the rationale for this limitation would be helpful. Your PR description I think explains it pretty well:

```suggestion
            // The LimitedDistinctAggregation rule should be applied before the EnforceDistribution rule
            // As that rule may inject other operations in between the different AggregateExecs.
            // Applying the rule early means only directly-connected AggregateExecs must be examined.
            
```

Reply:
> I think adding the rationale for this limitation would be helpful. Your PR description I think explains it pretty well:

Made the suggested change.